File: frontend/types/history.ts
// Progumi/frontend/types/history.ts
export type StudiedTerm = {
  term: string;      // 用語名
  is_correct: boolean; // 正誤判定
  category: string; // カテゴリ名
};

export type HistoryEntry = {
  quizResultId: string;
  createdAt: string; // quiz_results.created_at
  totalQuestions: number; // quiz_results.total_questions
  correctAnswers: number; // quiz_results.correct_answers
  score: number; // quiz_results.score
  studiedTerms: StudiedTerm[]; // 用語名と正誤情報のセット
};

File: backend/app/schemas/history.py
# Progumi/backend/app/schemas/history.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime


# 用語名と正誤情報をセットで返す型
class StudiedTermResponse(BaseModel):
    term: str        # 用語名
    is_correct: bool # 正誤判定
    category: str # カテゴリ名

    class Config:
        from_attributes = True


class HistoryEntryResponse(BaseModel):
    quizResultId: UUID
    createdAt: datetime
    totalQuestions: int
    correctAnswers: int
    score: int
    studiedTerms: list[StudiedTermResponse]  # 用語名と正誤情報のセット

    class Config:
        from_attributes = True


class QuizResultQuestionCreate(BaseModel):
    quiz_question_id: str  # フロントのidがundefinedになるケースの対策
    is_correct: bool
    selected_choice: int


class QuizHistoryCreate(BaseModel):
    score: int
    total_questions: int
    questions: list[QuizResultQuestionCreate]
